service/device/registry/test/src/test/resources/features/DeviceRegistry.feature

```diff
@@ -16,237 +16,209 @@ Feature: Device Registry CRUD tests
+        Kapua should not return any error.
+
+        Given A device named "TestDevice"
+        Given I create 100 randomly named devices with BIOS version "1.1.0"
+        Given I create 201 randomly named devices with BIOS version "1.2.0"
+        Given I create 98 randomly named devices with BIOS version "1.3.0"
+        When I query for devices with BIOS version "1.1.0"
+        Then I find 100 devices
+        When I query for devices with BIOS version "1.3.0"
+        Then I find 98 devices
+        When I query for devices with BIOS different from "1.2.0"
+        Then I find 201 devices
+        When I query for devices with Client Id "TestDevice"
+        Then I find 1 device
+
+    Scenario: Count devices with a specific BIOS version
+        It must be possible to count devices based on arbitrary rules.
+        To this end several devices are created with different BIOS version.
+        A device count based on the BIOS version value is performed. Only
+        the number of devices that match the specified BIOS version is
+        returned.
+
+        Given I create 15 randomly named devices with BIOS version "1.1.0"
+        Given I create 25 randomly named devices with BIOS version "1.2.0"
+        Given I create 35 randomly named devices with BIOS version "1.3.0"
+        When I count devices with BIOS version "1.2.0"
+        Then I count 25
+
+    Scenario: Try to update the device client ID
+        The Client ID of a device cannot be changed after creation. Any attempt to
+        after this ID must be silently ignored. No exception should be raised.
+
+        Given A device named "TestDevice"
+        When I update the device clientID from "TestDevice" to "NewClientId"
+        Then The client ID was not changed
+
+    Scenario: Update a non existing device
+        An attempt to update a non existing device should raise an exception.
+
+        Given A device named "TestDevice"
+        And I expect the exception "KapuaEntityNotFoundException" with the text "The entity of type device with id/name"
+        When I update a device with an invalid ID
+        Then An exception was thrown
+
+    Scenario: Delete an existing device from the registry
+        It must be possible to delete a device from the registry. To this
+        end a test device is created and subsequently deleted.
+        A search for this device should yield a null but no exception.
+
+        Given A device named "TestDevice"
+        When I delete the device with the clientId "TestDevice"
+        Then There is no device with the client ID "TestDevice"
+
+    Scenario: Try to delete a non existing device from the registry
+        If a user tries to delete a non existing device from the registry an
+        exception must be raised.
+
+        Given I expect the exception "KapuaEntityNotFoundException" with the text "The entity of type DeviceImpl with id/name"
+        When I delete a device with random IDs
```

Comment:
Can you please change the name of this step to: "When I delete a device with nonexisting ID"